documentation/Cohen Hubbard Oesterwinter.tex

```diff
@@ -0,0 +1,166 @@
+\documentclass[10pt, a4paper, twoside]{basestyle}
+
+\usepackage[Mathematics]{semtex}
+\usepackage{chngcntr}
+\counterwithout{equation}{section}
+
+%%%% Shorthands.
+
+%%%% Title and authors.
+
+\title{On a Formula by Cohen, Hubbard and Oesterwinter}
+\date{\printdate{2021-03-13}}
+\author{Robin~Leroy (eggrobin) \& Pascal~Leroy (phl)}
+\begin{document}
+\maketitle
+\begin{sloppypar}
+\noindent
+This document proves and generalizes a formula given in \cite{CohenHubbardOesterwinter1973} to compute the velocity of a body in the context of numerical integration of $n$-body systems.
+\end{sloppypar}
+
+\subsection*{Statement}
+\cite{CohenHubbardOesterwinter1973} states that
+\begin{quotation}
+the formula for a velocity component [is] of the form:
+\begin{equation}
+\TimeDerivative{x}_{n + 1} = \frac{1}{h}\pa{x_n - x_{n - 1} + h^2 \sum{i = 0}[12] \gb_i \SecondTimeDerivative{x}_{n-i}}
+\label{eqncho}
+\end{equation}
+\end{quotation}
+They then proceed to tabulate explicit values for the (rational) coefficients $\gb_i$ without explaining how they are computed.  This makes it impossible to use this formula for integrators of a different order or to construct similar formulæ for slightly different purposes.
+
+\subsection*{Finite difference formulæ}
+In this section we prove a backward difference formula that makes it possible to compute an approximation of any derivative of a function on an equally-spaced grid at any desired order.  We then derive a corollary based on the second derivative $f\dersecond\of{x}$ that is useful for the following sections. 
+\begin{lemma}
+Given a sufficient regular function $f\of{x}$ and two integers $m ≤ n$, there exists a family of rational numbers $\gl^{m}_{n,\gn}$, with $0 ≤ \gn ≤ n$, such that:
+
+\begin{equation}
+f^{(m)}\of{x_0} = \frac{\pa{-1}^m}{h^m} \sum{\gn = 0}[n]{\gl^{m}_{n,\gn} f\of{x_0 - \gn h}} + \BigO\of{h^{n - m + 1}} 
+\label{fornbergbackward}
+\end{equation}
+Furthermore, the $\gl^{m}_{n,\gn}$ are independent from $x_0$ and $h$.
+
+\begin{proof}
+\cite{Fornberg1988} gives finite difference formulæ for any order and for kernels of arbitrary size.  Given a family $\VectorSymbol{\ga} = \pa{\ga_0, \ga_1, ..., \ga_N}$ of points (which may not be equidistant), the $m$-th derivative at any point $x_0$ may be approximated to order $n - m + 1$ as:
+\[
+f^{(m)}\of{x_0} ≅ \sum{\gn = 0}[n]{\gd^{m}_{n,\gn}\of{\VectorSymbol{\ga}} f\of{\ga_\gn}}
+\]
+where $m ≤ n$ and the coefficients $\gd^{m}_{n,\gn}$ are dependent on $\VectorSymbol{\ga}$ but independent of $f$.
+
+If the $\VectorSymbol{\ga}$ are equally spaced with step $-h$ (\idest, $\ga_\gn = x_0 - \gn h$) then we can write a backward formula as follows:
+\begin{equation}
+f^{(m)}\of{x_0} = \sum{\gn = 0}[n]{\gd^{m}_{n,\gn}\of{\VectorSymbol{\ga}} f\of{x_0 - \gn h}} + \BigO\of{h^{n - m + 1}}
+\label{fornbergbackwardpreliminary}
+\end{equation}
+
```

Comment:
Avoid single-line paragraphs starting immediately after display maths; the new paragraph brings no clarity, and the repeated indent makes it look like the margin changes: remove this blank line.

Reply:
That paragraph change is still there and still looks weird (especially since it starts the page).